src/Response/Diagnostic/PerformanceData/PerformanceDataLine.php

```diff
@@ -0,0 +1,441 @@
+    {
+        if ($valueWithUnit === static::VALUE_UNKNOWN) {
+            $this->setValue(static::VALUE_UNKNOWN);
+            $this->setUnitOfMeasurement('');
+        } else {
+            $pattern = '/^\s*([-0-9.]+)\s*([A-Z%]+)?\s*$/i';
+
+            if (preg_match($pattern, $valueWithUnit, $matches) !== 1) {
+                throw new \InvalidArgumentException(
+                    'Invalid value with unit of measurement supplied: '
+                    . var_export($valueWithUnit, true)
+                );
+            }
+
+            $value = $matches[1];
+
+            // Set the value.
+            if (strpos($value, '.') >= 0) {
+                $this->setValue((float)$value);
+            } else {
+                $this->setValue((int)$value);
+            }
+
+            // Set the unit of measurement.
+            if (count($matches) === 3) {
+                $uom = $matches[2];
+
+                $allowedUnits = array_map(
+                    'strtolower',
+                    static::getAllowedUnitsOfMeasurement()
+                );
+
+                if (!in_array(strtolower($uom), $allowedUnits, true)) {
+                    throw new \InvalidArgumentException(
+                        'Invalid unit of measurement supplied: '
+                        . var_export($uom, true)
+                    );
+                }
+
+                $this->setUnitOfMeasurement($uom);
+            } else {
+                $this->setUnitOfMeasurement('');
+            }
+        }
+    }
+
+    /**
+     * Create a text version of the performance data line.
+     *
+     * @return string
+     */
+    public function toText()
+    {
+        $label = $this->getLabel();
+        $value = $this->getValue();
+        $uom = $this->getUnitOfMeasurement();
+        $warning = $this->getWarning();
+        $critical = $this->getCritical();
+        $min = $this->getMinimum();
+        $max = $this->getMaximum();
+
+        // Double single quote to escape single quote.
+        $label = str_replace("'", "''", $label);
+
+        $warning = ($warning === null) ? 'null' : $warning;
+        $critical = ($critical === null) ? 'null' : $critical;
+
+        // Not necessary for percentages to supply a min and max.
+        $min = ($uom === '%' || $uom === '') ? static::VALUE_UNKNOWN : $min;
+        $max = ($uom === '%' || $uom === '') ? static::VALUE_UNKNOWN : $max;
+
+        return "'{$label}'={$value}{$uom};{$warning};{$critical};{$min};{$max}";
+    }
+
+    /**
+     * Magic method which returns the text version when the object is being cast
+     * to string.
+     *
+     * @return string
+     */
+    public function __toString()
+    {
+        return $this->toText();
+    }
+
+    /**
+     * Get the allowed units of measurement.
+     *
+     * @return string[]
+     */
+    private static function getAllowedUnitsOfMeasurement()
+    {
+        return array('s', 'us', 'ms', '%', 'KB', 'MB', 'TB', 'c');
```

Comment:
I had the same thoughts when writing the code, but since these values are only being used for validation, I don't think there is added value in creating constants.